client/starwhale/utils/venv.py

```diff
@@ -107,12 +108,18 @@ def _do_pip_install_req(
     _hosts: t.List[str] = [_env.get("SW_PYPI_TRUSTED_HOST", "")]
     _index: str = _env.get("SW_PYPI_INDEX_URL", "")
 
+    list_to_str: t.Callable[[t.Union[str, list]], str] = (
+        lambda x: " ".join(x) if isinstance(x, list) else x
+    )
+
+    _pip_index_url = list_to_str(pip_config.get("index_url", ""))
     if _index:
-        _extra_index.append(pip_config.get("index_url", ""))
+        _extra_index.append(_pip_index_url)
     else:
-        _index = pip_config.get("index_url", "")
-    _extra_index.extend(pip_config.get("extra_index_url", []))
-    _hosts.extend(pip_config.get("trusted_host", []))
+        _index = _pip_index_url
+
+    _extra_index.append(list_to_str(pip_config.get("extra_index_url", [])))
+    _hosts.append(list_to_str(pip_config.get("trusted_host", [])))
```

Comment:
Do we support specify string var to these two fields?

Reply:
ok